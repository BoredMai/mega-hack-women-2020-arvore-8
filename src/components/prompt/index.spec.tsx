import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Prompt from '.';

describe('<Prompt>', () => {
  const props = {
    onPromptUpdate: jest.fn().mockName('onPromptUpdate'),
    prompt: {
      key: 'promptKey',
      type: 'input',
    },
  };
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Prompt {...props} />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('returns null when there is no prompt', () => {
    wrapper = shallow(<Prompt {...props} prompt={null} />) as ShallowWrapper;

    expect(wrapper.getElement()).toBeNull();
  });

  it('calls onPromptUpdate with input value on confirm', () => {
    const inputValue = 'inputValue';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: inputValue } });

    wrapper.find('button').simulate('click');
    expect(props.onPromptUpdate).toHaveBeenCalledWith(
      props.prompt.key,
      inputValue,
    );
  });
});
