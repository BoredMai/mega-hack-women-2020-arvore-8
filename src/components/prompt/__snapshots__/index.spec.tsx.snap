// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Prompt> renders properly 1`] = `
<div
  className="prompt"
>
  <input
    onChange={[Function]}
    type="input"
  />
  <button
    onClick={[Function]}
  >
    OK
  </button>
</div>
`;
